streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalProcessorContextImpl.java

```diff
@@ -47,23 +50,48 @@ public StateStore getStateStore(final String name) {
     @SuppressWarnings("unchecked")
     @Override
     public <K, V> void forward(final K key, final V value) {
+        forward(key, value, SEND_TO_ALL);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <K, V> void forward(final K key, final V value, final To to) {
+        toInternal.update(to);
+        if (toInternal.hasTimestamp()) {
+            recordContext.setTimestamp(toInternal.timestamp());
+        }
         final ProcessorNode previousNode = currentNode();
         try {
-            for (final ProcessorNode child : (List<ProcessorNode<K, V>>) currentNode().children()) {
-                setCurrentNode(child);
-                child.process(key, value);
+            final List<ProcessorNode<K, V>> children = (List<ProcessorNode<K, V>>) currentNode().children();
+            final String sendTo = toInternal.child();
+            if (sendTo != null) {
+                final ProcessorNode child = currentNode().getChild(sendTo);
+                if (child == null) {
+                    throw new StreamsException("Unknown downstream node: " + sendTo + " either does not exist or is not" +
+                        " connected to this processor.");
+                }
+                forward(child, key, value);
+            } else {
+                if (children.size() == 1) {
```

Comment:
I copied this from `ProcessorContextImpl#forward()` -- we added it there are "optimization" to avoid creating the iterator -- might be pre-mature optimization to be fair.

I don't feel strong about it, however, I would prefer to keep both classed "in sync". Thus, if we remove it here, I think we should also remove it from there?

Throughts?

Reply:
Ok. Will simplify this.